MetaMask/metamask-extension
# Convert store.js to typescript and redux toolkit

## Explanation
Switches the store from using the redux 'createStore' method to the configureStore method from reduxjs/toolkit. The toolkit configureStore methods has a lot of good features and type inference and we want to generally adopt more of the reduxjs/toolkit features. 

There are two changes that needed to be made in order to get this passing in E2E:

By default configureStore loads two middleware that we did not previously use:
1. serialzableCheck middleware -- checks if the state keys are serializable thus can be persisted in browser state for rehydration.
2. immutableCheck middleware -- checks that state is immutable and not mutated.

These two middleware should be incrementally adopted, but for now they are disabled, and in the case of immutableCheck only enabled in dev mode. I had to specify a higher warnAfter level because our actions take longer than expected to check (32ms -> 80ms) we can lower that number overtime. 

While not *technically* part of this work, the inclusion of a few baseline entries in the state tree type will  aid the action typescript conversion pr. 

** In addition ** 
I made a minor change to the metamask duck to allow typescript to infer the shape of its state, though this should also have no functional impact. 

## Manual Testing Steps
1. Pull branch
2. `yarn`
3. `yarn start`
4. etiher with or without redux devtools, perform a number of actions
5. confirm no change in behavior.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
